src/ml-search.service.js

```diff
@@ -1128,6 +1121,37 @@
     },
 
     /**
+     * Retrieves values or tuples from 1-or-more lexicons
+     * @method MLSearchContext#values
+     *
+     * @param {String} name - the name of a `value-option` definition
+     * @param {Object} [params] - URL params
+     * @param {Object} [options] - search options, used in a combined query
+     * @return {Promise} a promise resolved with values
+     */
+    values: function values(name, params, options) {
+      // TODO: what other conditions
+      if ( !options && params && params.options) {
+        options = params;
+        params = null;
+      }
+
+      params = params || {};
+      params.start = params.start || 1;
+      params.limit = params.limit || 20;
+
+      if ( !options ) {
+        params.options = this.options.queryOptions;
+      }
+
+      return this.getCombinedQuery(false)
+      .then(function(combined) {
+        combined.options = options;
```

Comment:
How about getting stored options, checking whether the values are defined, and if not look for facet constraint that could be used to dynamically generate values config (like `showMoreFacets` is doing)?